app/Http/Controllers/App/UsersController.php

```diff
@@ -36,58 +48,176 @@ public function add(Request $request)
 
     public function browse(Request $request)
     {
+        return self::json([], 501, ['message' => 'not yet implemented <3']);
         // TODO check privileges
-        $users = User::with('AdserverWallet')->whereNull('deleted_at')->get();
+        $users = User::with('AdserverWallet')->get();
 
         return self::json($users->toArrayCamelize());
     }
 
     public function delete(Request $request, $user_id)
     {
+        return self::json([], 501, ['message' => 'not yet implemented <3']);
         // TODO check privileges
         // TODO reset email
         // TODO process
-        $user = User::whereNull('deleted_at')->findOrFail($user_id);
-        $user->deleted_at = new \DateTime();
-        $user->save();
+        $user = User::findOrFail($user_id);
+        $user->delete();
 
-        return self::json(['message' => 'successfully deleted'], 200);
+        return self::json([], 204);
     }
 
-    public function edit(Request $request, $user_id)
+    public function edit(Request $request)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this function 6, down to the maximum allowed 3. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=php%3AS1142)